wagtail_streamfield_migration_toolkit/test/tests/test_migrations.py

```diff
@@ -0,0 +1,232 @@
+        )
+
+        for instance in instances:
+            old_revisions = self.revisions[instance.id]
+            for old_revision, new_revision in zip(
+                old_revisions, instance.revisions.all().order_by("id")
+            ):
+                old_content = json.loads(old_revision.content["content"])
+                new_content = json.loads(new_revision.content["content"])
+                for old_block, new_block in zip(old_content, new_content):
+                    self.assertEqual(old_block["id"], new_block["id"])
+                    if old_block["type"] == "char1":
+                        self.assertEqual(new_block["type"], "renamed1")
+                    else:
+                        self.assertEqual(old_block["type"], new_block["type"])
+
+    def _test_always_migrate_live_and_latest_revisions(self):
+        revisions_from = datetime.datetime.now() + datetime.timedelta(days=2)
+        self.apply_migration(revisions_from=revisions_from)
+
+        instances = self.model.objects.all().annotate(
+            raw_content=Cast(F("content"), JSONField())
+        )
+
+        for instance in instances:
+            old_revisions = self.revisions[instance.id]
+            for old_revision, new_revision in zip(
+                old_revisions, instance.revisions.all().order_by("id")
+            ):
+                is_latest_or_live = (
+                    old_revision.id == instance.live_revision_id
+                    or old_revision.id == instance.latest_revision_id
+                )
+                old_content = json.loads(old_revision.content["content"])
+                new_content = json.loads(new_revision.content["content"])
+                for old_block, new_block in zip(old_content, new_content):
+                    self.assertEqual(old_block["id"], new_block["id"])
+                    if is_latest_or_live and old_block["type"] == "char1":
+                        self.assertEqual(new_block["type"], "renamed1")
+                    else:
+                        self.assertEqual(old_block["type"], new_block["type"])
+
+    def _test_migrate_revisions_from_date(self):
+        revisions_from = datetime.datetime.now() - datetime.timedelta(days=2)
+        self.apply_migration(revisions_from=revisions_from)
+
+        instances = self.model.objects.all().annotate(
+            raw_content=Cast(F("content"), JSONField())
+        )
+
+        for instance in instances:
+            old_revisions = self.revisions[instance.id]
+            for old_revision, new_revision in zip(
+                old_revisions, instance.revisions.all().order_by("id")
+            ):
+                is_latest_or_live = (
+                    old_revision.id == instance.live_revision_id
+                    or old_revision.id == instance.latest_revision_id
+                )
+                is_after_revisions_from = (
+                    old_revision.created_at.timestamp() > revisions_from.timestamp()
```

Comment:
Are these are both `datetime` instances? If so they can be compared directly without casting them to a timestamp

Reply:
I think this was because one was tz aware and the other wasn't and hence couldnt compare directly. Anyway I changed it to tz aware values now. (also got rid of some warnings)